packages/pharos/src/components/coach-mark/pharos-coach-mark.scss

```diff
@@ -0,0 +1,199 @@
+  --coach-mark-border-color: var(--pharos-color-marble-gray-base);
+  --coach-mark-border-width: 1px;
+  --coach-mark-text-size: var(--pharos-font-size-small);
+  --coach-mark-heading-color: var(--pharos-color-marble-gray-20);
+  --coach-mark-text-color: var(--pharos-color-marble-gray-20);
+  --coach-mark-text-spacing: var(--pharos-spacing-1-x);
+  --coach-mark-pip-size: 10px;
+
+  display: block;
+  height: 0;
+  width: 0;
+}
+
+:host([variant='dark']) {
+  --coach-mark-background-color: var(--pharos-button-color-base-overlay-background-base);
+  --coach-mark-border-color: var(--coach-mark-background-color);
+  --coach-mark-heading-color: var(--pharos-color-white);
+  --coach-mark-text-color: var(--pharos-color-marble-gray-80);
+}
+
+.coach-mark {
+  animation-duration: 1000ms; /* Delay duration of 0ms, fade duration of --pharos-transition-duration-default */
+  animation-name: fadein;
+  animation-timing-function: ease-in-out;
+
+  &[aria-hidden='true'] {
+    display: none;
+    opacity: 0;
+  }
+
+  &.delay-short {
+    animation-duration: 1600ms; /* Delay duration of 600ms, fade duration of --pharos-transition-duration-default */
+    animation-name: fadein-short;
+  }
+
+  &.delay-long {
+    animation-duration: 3000ms; /* Delay duration of 2000ms, fade duration of --pharos-transition-duration-default */
+    animation-name: fadein-long;
+  }
+}
+
+.coach-mark__content {
+  position: relative;
+  background-color: var(--coach-mark-background-color);
+  border: var(--coach-mark-border-width) solid var(--coach-mark-border-color);
+  padding: var(--pharos-spacing-one-half-x) calc(24px + var(--pharos-spacing-one-and-a-half-x))
+    var(--pharos-spacing-one-half-x) var(--pharos-spacing-three-quarters-x);
+  min-width: 100px;
+  width: fit-content;
+  font-size: var(--coach-mark-text-size);
+  color: var(--coach-mark-text-color);
+  line-height: var(--coach-mark-text-spacing);
+
+  &::before,
+  &::after {
+    content: '';
+    background-color: var(--coach-mark-background-color);
+    position: absolute;
+    top: var(--pip-top);
+    left: var(--pip-left);
+    height: var(--coach-mark-pip-size);
+    width: var(--coach-mark-pip-size);
+    transform: translate(-50%, -50%) rotate(45deg);
+    z-index: 1;
+  }
+
+  &::after {
+    --coach-mark-border-width-x2: calc(var(--coach-mark-border-width) * 2);
+
+    background-color: var(--coach-mark-border-color);
+    height: calc(var(--coach-mark-pip-size) + var(--coach-mark-border-width-x2));
+    width: calc(var(--coach-mark-pip-size) + var(--coach-mark-border-width-x2));
+    z-index: -1;
+  }
+
+  pharos-heading {
+    color: var(--coach-mark-heading-color);
+  }
+
+  .coach-mark__close {
+    position: absolute;
+    top: 0;
+    right: 0;
+  }
+}
+
+.coach-mark__wrapper {
+  height: fit-content;
+  width: fit-content;
+
+  &.coach-mark-side__right {
+    transform: translateY(-50%);
+
+    .coach-mark__content {
+      --pip-left: 0;
+    }
+  }
```

Comment:
There's a fair amount of nesting going on here; the value of BEM is generally not to need to nest things because everything is similar/same specificity. Can this whole section be flattened out a bit?

Reply:
I was able to reduce the amount of nesting, which was also helped by the addition of `floating-ui` mentioned below